### PR TITLE
Feature/widget descriptions and configuration help

### PR DESCRIPTION
**Pull request by JBartscher**
# Adds tooltips to widgets and improves the clarity of errors

* catch errors instead of passing them to the adf-widget framework to display meaningful errors instead of "please configure your widget"
* change package manager from npm to yarn
* Add tooltips to inputs
* replace package-lock.json with yarn.lock
* rebuild dist folder
* update vulnerabilities in dependencies that had a fix in `npm audit`. This widget should now work with node >=10.23.
* Update gulp to v4, therefore rewrite of most of the gulp.tasks.
* update CHANGELOG.md
* update Readme with instructions on how to develop localy

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,6 +41,27 @@ Each goal can be used as parameter for the gulp command.
...
+* At the end of `easyRedmineService.js` and `redmineService.js` remove the comments around
+  ```js
+  config(function($httpProvider) {
+  $httpProvider.defaults.headers.common['Authorization'] = 'Basic ' + 'BASE64HERE=';
+  })
+  ```
+  and replace "BASE64HERE" with a base64 hashed combination of `username:password`.
+  
+
+* Open a terminal and start an unsafe chrome-tab with (otherwise you will get CORS errors) with this line:
```
Comment (jsprey) [inline README.md:57]: Is maybe a chrome extension, like a CORS unblocker, also sufficient?

File: src/main/chart/edit/edit.html [hunk 2]
```diff
@@ -1,27 +1,55 @@
...
-    <select name="project" id="project" class="form-control" ng-model="vm.config.project" ng-change="vm.checkUpdates()" ng-required="true">
+    <span class="grey glyphicon glyphicon-info-sign ng-scope"
+          tabindex="0"
+          ng-focus="projectTooltipIsOpen = !projectTooltipIsOpen"
+          tooltip-trigger="outsideClick"
+          tooltip-is-open="projectTooltipIsOpen"
+          ng-mouseenter="projectTooltipIsOpen = !projectTooltipIsOpen"
+          uib-tooltip="Choose a project"
+          tooltip-placement="top"
+          aria-label="Choose a project"></span>
```
Comment (jsprey) [inline src/main/chart/edit/edit.html]:
```suggestion
          aria-label="Choose a project. If you don't see any project make sure you have correct access rights in redmine. "></span>
```

File: src/main/chart/edit/edit.html [hunk 3]
```diff
@@ -60,22 +105,45 @@
 
   <div class="form-group">
     <input type="checkbox" name="showIdeal" ng-model="config.showIdeal"> Show ideal line
+    <span class="grey glyphicon glyphicon-info-sign ng-scope"
+          tabindex="0"
+          ng-focus="showIdealTooltipIsOpen = !showIdealTooltipIsOpen"
+          tooltip-trigger="outsideClick"
+          tooltip-is-open="showIdealTooltipIsOpen"
+          ng-mouseenter="showIdealTooltipIsOpen = !showIdealTooltipIsOpen"
+          uib-tooltip="choose a tracker type like bug, feature or support"
```
Comment (jsprey) [inline src/main/chart/edit/edit.html]:
```suggestion
          uib-tooltip="Shows an additional line in the chart representing the ideal progress of the burndown progress."
```

File: src/main/chart/edit/edit.html [hunk 4]
```diff
@@ -60,22 +105,45 @@
...
+          tooltip-placement="top"
+          aria-label="choose a tracker type like bug, feature or support"></span>
```
Comment (jsprey) [inline src/main/chart/edit/edit.html]:
```suggestion
          aria-label="optionally, show an additional line in the chart representing the ideal progress of the burndown progress"></span>
```

File: src/main/chart/edit/edit.html [hunk 5]
```diff
@@ -60,22 +105,45 @@
 
   <div class="form-group">
     <input type="checkbox" name="showIdeal" ng-model="config.showIdeal"> Show ideal line
```
Comment (jsprey) [inline src/main/chart/edit/edit.html:107]:
```suggestion
    <input type="checkbox" name="showIdeal" ng-model="config.showIdeal"> Show ideal burndown line
```

File: src/main/chart/edit/editChartController.js [hunk 6]
```diff
@@ -130,10 +131,14 @@ function createEditChartController(vm, projects, config, chartDataService, apiSe
   }
 }
 
-angular.module('adf.widget.redmine').controller('editChartController', function (projects, config, chartDataService, redmineService) {
+angular.module('adf.widget.redmine').controller('editChartController', function (projects, config, chartDataService, redmineService, $scope, $sce) {
+  $scope.showIdealTooltip = $sce.trustAsHtml('<b>OPTIONAL</b><br>If checked displays an ideal line on the chart');
+  $scope.assignedToTooltip = $sce.trustAsHtml('Get issues which are assigned to the given user ID. <b>me</b> (but not <b>&lt;me&gt;</b> or <b>&lt;&lt;me&gt;&gt;</b>) can be used instead an ID to fetch all issues from the logged in user. Leave empty if you want to see all issues.');
```
Comment (jsprey) [inline src/main/chart/edit/editChartController.js:135]: We should move it to the html files not to the controller

File: src/main/chart/edit/editChartController.js [hunk 7]
```diff
@@ -130,10 +131,14 @@ function createEditChartController(vm, projects, config, chartDataService, apiSe
...
   return createEditChartController(this, projects, config, chartDataService, redmineService);
 });
 
-angular.module('adf.widget.easyredmine').controller('easyEditChartController', function (projects, config, easyChartDataService, easyRedmineService) {
+angular.module('adf.widget.easyredmine').controller('easyEditChartController', function (projects, config, easyChartDataService, easyRedmineService,  $scope, $sce) {
+  $scope.showIdealTooltip = $sce.trustAsHtml('<b>OPTIONAL</b><br>If checked displays an ideal line on the chart');
```
Comment (jsprey) [inline src/main/chart/edit/editChartController.js]: redudnant

File: src/main/issues/edit/editIssuesController.js [hunk 8]
```diff
@@ -23,11 +24,13 @@ function createEditIssueController(vm, projects, customQueries, config){
 }
 
 angular.module('adf.widget.redmine')
-  .controller('editIssuesController', function (projects, customQueries, config) {
+  .controller('editIssuesController', function (projects, customQueries, config, $scope, $sce) {
+    $scope.customQueryTooltip = $sce.trustAsHtml('Select a custom query. Querys can be defined in Redmine.');
```
Comment (jsprey) [inline src/main/issues/edit/editIssuesController.js:28]: hmm, i dont like it

File: src/main/services/apiService.js [hunk 9]
```diff
@@ -8,12 +8,17 @@ function ApiService($http, apiEndpoint, apiEndpointRedirect, $q) {
 }
 
 ApiService.prototype.request = function (param) {
+
+  //auth = new Buffer(auth).toString('base64');
```
Comment (jsprey) [inline src/main/services/apiService.js]: This should be removed

**Review by jsprey — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (JBartscher) [reply README.md:57]: Iam afraid not

**Review by jsprey — APPROVED** [review]

**Pull request merged**